### PR TITLE
contrib: deposit extension

### PR DESCRIPTION
**Pull request by omelkonian**
- Adds extension for Invenio-Deposit. (closes CERNDocumentServer/cds#291)
- Adds test for coverage.

Signed-off-by: Orestis Melkonian melkon.or@gmail.com

### CODE REVIEW

**Comment by omelkonian** [comment]
@egabancho Let me know if you want me to move extension registration from `config.py` to entrypoints from `setup.py`.

**Comment by egabancho** [comment]
I think it is better if we use entry points, i.e. `invenio_sse.extensions`

File: invenio_sse/contrib/deposit/ext.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        :param app: An instance of :class:`flask.Flask`.
+        """
+        self.init_config(app)
+        blueprint = create_blueprint(
+            app.config['DEPOSIT_REST_ENDPOINTS']
+        )
+
+        app.register_blueprint(blueprint)
+
+    def init_config(self, app):
```
Comment (egabancho) [inline invenio_sse/contrib/deposit/ext.py]: This part probably you don't need it, it is already done by the "main" extension

File: invenio_sse/ext.py [hunk 2]
```diff
@@ -42,6 +42,12 @@ def __init__(self, app):
         """Initialize state."""
         self.app = app
         self._redis = StrictRedis.from_url(app.config['SSE_REDIS_URL'])
+        self.extensions = app.config['SSE_EXTENSIONS']
```
Comment (egabancho) [inline invenio_sse/ext.py]: I would extract the registration part to a different method, see for example [invenio-pidstore](https://github.com/inveniosoftware/invenio-pidstore/blob/master/invenio_pidstore/ext.py#L82) and maybe add the name of the entry point group to the `init_app`

File: invenio_sse/contrib/deposit/rest.py [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    def get(self, pid, record):
+        """Initiate SSE connection.
+
+        Permission required: `read_permission_factory`.
+
+        :param pid: Pid object (from url).
+        :param record: Record object resolved from the pid.
+        :returns: the established SSE channel.
+        """
+        from invenio_sse import current_sse
```
Comment (egabancho) [inline invenio_sse/contrib/deposit/rest.py]: Can you put the import on the top?

File: invenio_sse/cli.py [hunk 4]
```diff
@@ -1,7 +1,7 @@
 # -*- coding: utf-8 -*-
 #
 # This file is part of Invenio.
-# Copyright (C) 2015, 2016 CERN.
```
Comment (egabancho) [inline invenio_sse/cli.py:4]: 👏

File: invenio_sse/ext.py [hunk 5]
```diff
@@ -38,10 +39,27 @@
...
         self.app = app
         self._redis = StrictRedis.from_url(app.config['SSE_REDIS_URL'])
+        self.integrations = {}
+
+        if entry_point_group:
+            self.load_integration(entry_point_group)
+
+    def register_integration(self, integration_id, integration):
+        """Register an SSE integration with another module."""
+        assert integration_id not in self.integrations
```
Comment (egabancho) [inline invenio_sse/ext.py]: I think entry points already does this for you...

**Comment by omelkonian** [comment]
The `Flask-Security-Fork` dependency doesn't seem to resolve the `ContextualVersionConflict`.

**Comment by omelkonian** [comment]
@egabancho ready

**Pull request merged**